code/projects/app/src/app/shared/image-comparison/image-comparison.service.ts

```diff
@@ -0,0 +1,35 @@
+import { computed, Injectable, signal } from "@angular/core";
+
+export type Side = "left" | "right";
+
+@Injectable({
+  providedIn: "root",
+})
+export class ImageComparisonService {
+  public readonly left = signal<ImageData | undefined>(undefined);
+  public readonly right = signal<ImageData | undefined>(undefined);
+
+  public readonly images = computed(() => {
+    return [this.left(), this.right()];
+  });
```

Comment:
```suggestion
  public readonly images = computed(() => [this.left(), this.right()] as const);
```
**nit:** We can add `as const` to infer it as tuple instead of an array.